kzg_prover/src/chips/range/range_check.rs

```diff
@@ -1,113 +1,112 @@
+use crate::chips::range::utils::decompose_fp_to_byte_pairs;
 use halo2_proofs::arithmetic::Field;
 use halo2_proofs::circuit::{AssignedCell, Region, Value};
 use halo2_proofs::halo2curves::bn256::Fr as Fp;
 use halo2_proofs::plonk::{Advice, Column, ConstraintSystem, Error, Expression, Fixed};
 use halo2_proofs::poly::Rotation;
-
 use std::fmt::Debug;
 
-use crate::chips::range::utils::decompose_fp_to_bytes;
-
-/// Configuration for the Range Check Chip
+/// Configuration for the Range Check u64 Chip
+/// Used to verify that an element lies in the u64 range.
 ///
-/// # Type Parameters
+/// # Fields
 ///
-/// * `N_BYTES`: Number of bytes in which the element to be checked should lie
+/// * `zs`: Four advice columns - contain the truncated right-shifted values of the element to be checked
 ///
-/// # Fields
+/// # Assumptions
 ///
-/// * `zs`: Advice columns - contain the truncated right-shifted values of the element to be checked
+/// * The lookup table `range_u16` is by default loaded with values from 0 to 2^16 - 1.
 ///
 /// Patterned after [halo2_gadgets](https://github.com/privacy-scaling-explorations/halo2/blob/main/halo2_gadgets/src/utilities/decompose_running_sum.rs)
 #[derive(Debug, Copy, Clone)]
-pub struct RangeCheckConfig<const N_BYTES: usize> {
-    zs: [Column<Advice>; N_BYTES],
+pub struct RangeCheckU64Config {
+    zs: [Column<Advice>; 4],
 }
 
-/// Helper chip that verfiies that the element witnessed in a given cell lies within a given range defined by N_BYTES.
-/// For example, Let's say we want to constraint 0x1f2f3f4f to be within the range N_BYTES=4.
+/// Helper chip that verfiies that the element witnessed in a given cell lies within the u64 range.
+/// For example, Let's say we want to constraint 0x1f2f3f4f5f6f7f8f to be a u64.
+/// Note that the lookup table `range` is by default loaded with values from 0 to 2^16 - 1.
 /// `z` is the advice column that contains the element to be checked.
 ///
-/// `z = 0x1f2f3f4f`
-/// `zs[0] = (0x1f2f3f4f - 0x4f) / 2^8 = 0x1f2f3f`
-/// `zs[1] = (0x1f2f3f - 0x3f) / 2^8 = 0x1f2f`
-/// `zs[2] = (0x1f2f - 0x2f) / 2^8 = 0x1f`
-/// `zs[3] = (0x1f - 0x1f) / 2^8 = 0x00`
+/// `z = 0x1f2f3f4f5f6f7f8f`
+/// `zs[0] = (0x1f2f3f4f5f6f7f8f - 0x7f8f) / 2^16 = 0x1f2f3f4f5f6`
+/// `zs[1] = (0x1f2f3f4f5f6 - 0xf5f6) / 2^16 = 0x1f2f3f4`
```

Comment:
`zs[0]`and `zs[1]` are missing `f` , It should be ends with `f` like:
= `0x1f2f3f4f5f6`
= `0x1f2f3f4f`